Realm/RLMBSON.mm

```diff
@@ -316,11 +317,12 @@ Bson RLMConvertRLMBSONToBson(id<RLMBSON> b) {
         case RLMBSONTypeNull:
             return util::none;
         case RLMBSONTypeBool:
-            return ((NSNumber *)b).boolValue;
+            return (bool)((NSNumber *)b).boolValue;
         case RLMBSONTypeDouble:
             return ((NSNumber *)b).doubleValue;
         case RLMBSONTypeBinary:
-            return (char**)((NSData *)b).bytes;
+            return std::vector<char>((char*)((NSData *)b).bytes,
+                                     ((char*)((NSData *)b).bytes) + (int)((NSData *)b).length);
```

Comment:
bad indent

Reply:
this one is weirdly correct, but the above one is wrong